tasks/orca_catalog_reporting/orca_catalog_reporting.py

```diff
@@ -0,0 +1,267 @@
+        end_timestamp: Cumulus createdAt end-time for date range to compare data.
+        page_index: The 0-based index of the results page to return.
+        db_connect_info: See requests_db.py's get_configuration for further details.
+    """
+    engine = shared_db.get_user_connection(db_connect_info)
+    granules = query_db(
+        engine,
+        provider_id,
+        collection_id,
+        granule_id,
+        start_timestamp,
+        end_timestamp,
+        page_index,
+    )
+
+    return {"anotherPage": len(granules) > PAGE_SIZE, "granules": granules[0:PAGE_SIZE]}
+
+
+@retry_operational_error()
+def query_db(
+    engine: Engine,
+    provider_id: Union[None, List[str]],
+    collection_id: Union[None, List[str]],
+    granule_id: Union[None, List[str]],
+    start_timestamp: Union[None, str],
+    end_timestamp: str,
+    page_index: int,
+) -> List[Dict[str, Any]]:
+    """
+
+    Args:
+        provider_id: The unique ID of the provider(s) making the request.
+        collection_id: The unique ID of collection(s) to compare.
+        granule_id: The unique ID of granule(s) to compare.
+        start_timestamp: Cumulus createdAt start time for date range to compare data.
+        end_timestamp: Cumulus createdAt end-time for date range to compare data.
+        page_index: The 0-based index of the results page to return.
+        engine: The sqlalchemy engine to use for contacting the database.
+    """
+    with engine.begin() as connection:
+        sql_results = connection.execute(
+            get_catalog_sql(),
+            [
+                {
+                    "provider_id": provider_id,
+                    "collection_id": collection_id,
+                    "granule_id": granule_id,
+                    "start_timestamp": start_timestamp,
+                    "end_timestamp": end_timestamp,
+                    "page_index": page_index,
+                    "page_size": PAGE_SIZE,
+                }
+            ],
+        )
+
+        granules = []
+        for sql_result in sql_results:
+            granules.append(
+                {
+                    "providerId": sql_result["provider_ids"],
+                    "collectionId": sql_result["collection_id"],
+                    "id": sql_result["id"],
+                    "createdAt": str(sql_result["cumulus_create_time"]).replace(
+                        "+00:00", "Z", 1
+                    ),
+                    "executionId": sql_result["execution_id"],
+                    "ingestDate": str(sql_result["ingest_time"]).replace(
+                        "+00:00", "Z", 1
+                    ),
+                    "lastUpdate": str(sql_result["last_update"]).replace(
+                        "+00:00", "Z", 1
+                    ),
+                    "files": sql_result["files"],
+                }
+            )
+        return granules
+
+
+def get_catalog_sql() -> text:
+    return text(
+        """
+SELECT
```

Comment:
@AndrewDorn I created a test database with a fair amount of data using the SQL in 
[test.txt](https://github.com/nasa/cumulus-orca/files/7445270/test.txt). The query is somewhat expensive that we might be able to tune. Maybe a card to revisit and tune it. We could also look at using CTE expressions to maybe make it easier to follow instead of all of the nested SQL. I am a little concerned about our paging mechanism as we are in essence calling for that same result set every time we go through a page. There maybe some better approaches that we can discuss.

Reply:
https://bugs.earthdata.nasa.gov/browse/ORCA-286